### PR TITLE
test: subgraph matchstick tests for access points and acl refactor

### PR DESCRIPTION
**Pull request by EmperorOrokuSaki**

### CODE REVIEW

File: subgraph/tests/matchstick/access-control/CollectionRoleChanged.test.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import { createNewCollectionRoleChanged, handleCollectionRoleChangedList, makeEventId, TOKEN_OWNER_ONE, TOKEN_OWNER_TWO } from '../helpers/utils';
+import { CollectionRoleChanged } from '../../../generated/FleekNFA/FleekNFA';
+
+describe('Collection Role Changed tests', () => {
+  beforeAll(() => {
+    // Collection Role Changed
+    let collectionRoleChangedList: CollectionRoleChanged[] = [];
+
+    collectionRoleChangedList.push(
+      createNewCollectionRoleChanged(0, 0, TOKEN_OWNER_ONE, true, TOKEN_OWNER_TWO) // Token Owner Two grants collection owner access to Token Owner One
```
Comment (jsonsivar) [inline subgraph/tests/matchstick/access-control/CollectionRoleChanged.test.ts:19]: The comment here is confusing.  How can a token owner grant collection owner perms to anyone?

File: subgraph/tests/matchstick/access-control/CollectionRoleChanged.test.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      );
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(2),
+        'status',
+        'false'
+      );
+    });
+  });
+});
```
Comment (jsonsivar) [inline subgraph/tests/matchstick/access-control/CollectionRoleChanged.test.ts:95]: I see that these all test the event entity, is there some non-event entity that these handlers update that we should also check?

File: subgraph/tests/matchstick/access-control/TokenRoleChanged.test.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        'status',
+        'true'
+      );
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(2),
+        'status',
+        'false'
+      );
+    });
```
Comment (jsonsivar) [inline subgraph/tests/matchstick/access-control/TokenRoleChanged.test.ts:141]: Similar to my previous comment, maybe here we can assert if the current token roles list in the entities is correct?

**Review by jsonsivar — APPROVED** [review]

File: subgraph/tests/matchstick/access-control/CollectionRoleChanged.test.ts (hunk 1 above)
Comment (EmperorOrokuSaki) [reply subgraph/tests/matchstick/access-control/CollectionRoleChanged.test.ts:19]: The name of the address const is token owner one/two. I see why that could be confusing. I will change them to user_one/two

File: subgraph/tests/matchstick/access-control/CollectionRoleChanged.test.ts (hunk 2 above)
Comment (EmperorOrokuSaki) [reply subgraph/tests/matchstick/access-control/CollectionRoleChanged.test.ts:95]: From our discussion on Discord:

To be discussed after #168

File: subgraph/tests/matchstick/access-control/TokenRoleChanged.test.ts (hunk 3 above)
Comment (EmperorOrokuSaki) [reply subgraph/tests/matchstick/access-control/TokenRoleChanged.test.ts:141]: From our discussion on Discord:

To be discussed after #168

**Pull request merged**